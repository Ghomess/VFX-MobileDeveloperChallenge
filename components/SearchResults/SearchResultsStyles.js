import {StyleSheet} from 'react-native';

export const searchResultsStyles = StyleSheet.create({
  container: {
    display: 'flex',
    width: '90%',
    alignContent: 'center',
    backgroundColor: '#1A237E',

    borderBottomRightRadius: 5,
    borderBottomLeftRadius: 5,
  },
  separator: {
    borderBottomWidth: 2,

    borderColor: '#424242',
  },
  itemContainer: {
    borderRadius: 5,
    padding: 15,
    paddingLeft: 15,
  },
  itemText: {
    color: '#ffffff',
    fontSize: 15,
  },
});
