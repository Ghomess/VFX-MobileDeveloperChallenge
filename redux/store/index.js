import {configureStore} from '@reduxjs/toolkit';
import stockSlice from '../reducers/stockSlice';
import pairSlice from '../reducers/pairSlice';
import loadingSlice from '../reducers/loadingSlice';

const store = configureStore({
  reducer: {
    ticker: stockSlice,
    pair: pairSlice,
    loading: loadingSlice,
  },
});

export default store;
