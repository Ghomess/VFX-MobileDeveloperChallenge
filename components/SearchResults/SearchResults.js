import React, {useEffect} from 'react';
import {FlatList, View} from 'react-native';
import {searchResultsStyles} from './SearchResultsStyles';
import {useDispatch} from 'react-redux';

import {useNavigation} from '@react-navigation/native';
import {changeLoadingSearchResults} from '../../redux/reducers/loadingSlice';
import {ItemComponent} from '../ItemComponent/ItemComponent';
import {ItemSeparatorComponent} from '../ItemSeparatorComponent/ItemSeparatorComponent';

export const SearchResults = ({data, setEnableList}) => {
  const navigation = useNavigation();
  const navigationState = navigation.getState();
  const currentScreen = navigationState.routes[navigationState.index].name;

  const dispatch = useDispatch();

  useEffect(() => {
    //if data from the fetch exists, it will stop loading
    if (data) {
      dispatch(changeLoadingSearchResults(false));
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data]);

  function renderItem({item}) {
    return (
      <ItemComponent
        name={item.name}
        setEnableList={setEnableList}
        screen={currentScreen}
      />
    );
  }

  return (
    <View style={searchResultsStyles.container}>
      <FlatList
        data={data}
        keyExtractor={item => item.id}
        renderItem={renderItem}
        ItemSeparatorComponent={ItemSeparatorComponent}
        testID="SearchResults.FlatList"
      />
    </View>
  );
};
