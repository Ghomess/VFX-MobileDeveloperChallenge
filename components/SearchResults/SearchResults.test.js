import React from 'react';

import {NavigationContainer} from '@react-navigation/native';
import {Provider} from 'react-redux';
import store from '../../redux/store';
import {render, waitFor} from '@testing-library/react-native';

import {SearchResults} from './SearchResults';
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    getState: () => ({
      routes: [{name: 'Currency'}],
      index: 0,
    }),
  }),
}));

const dataMocked = [
  {id: 0, name: 'T'},
  {id: 1, name: 'T.TRT'},
  {id: 2, name: 'T04.FRK'},
  {id: 3, name: 'T0A.FRK'},
  {id: 4, name: 'T0P.FRK'},
  {id: 5, name: 'T0T.FRK'},
  {id: 6, name: 'T18.FRK'},
  {id: 7, name: 'T1A.FRK'},
  {id: 8, name: 'T10G.LON'},
  {id: 9, name: 'T15B.FRK'},
];

test('renders correctly with data', async () => {
  const {getByTestId} = render(
    <Provider store={store}>
      <NavigationContainer>
        <SearchResults data={dataMocked} />
      </NavigationContainer>
    </Provider>,
  );
  const flatList = getByTestId('SearchResults.FlatList');
  expect(flatList).toBeDefined();
  await waitFor(() => {
    expect(flatList.props.data.length).toBe(dataMocked.length);
  });
});

test('renders correctly snapshot', async () => {
  const tree = render(
    <Provider store={store}>
      <NavigationContainer>
        <SearchResults data={dataMocked} />
      </NavigationContainer>
    </Provider>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
