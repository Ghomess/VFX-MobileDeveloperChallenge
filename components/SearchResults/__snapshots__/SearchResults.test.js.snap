// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly snapshot 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "backgroundColor": "#1A237E",
      "borderBottomLeftRadius": 5,
      "borderBottomRightRadius": 5,
      "display": "flex",
      "width": "90%",
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      [
        {
          "id": 0,
          "name": "T",
        },
        {
          "id": 1,
          "name": "T.TRT",
        },
        {
          "id": 2,
          "name": "T04.FRK",
        },
        {
          "id": 3,
          "name": "T0A.FRK",
        },
        {
          "id": 4,
          "name": "T0P.FRK",
        },
        {
          "id": 5,
          "name": "T0T.FRK",
        },
        {
          "id": 6,
          "name": "T18.FRK",
        },
        {
          "id": 7,
          "name": "T1A.FRK",
        },
        {
          "id": 8,
          "name": "T10G.LON",
        },
        {
          "id": 9,
          "name": "T15B.FRK",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    testID="SearchResults.FlatList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T.TRT
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T04.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T0A.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T0P.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T0T.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T18.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T1A.FRK
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T10G.LON
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomWidth": 2,
              "borderColor": "#424242",
            }
          }
        />
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderRadius": 5,
              "opacity": 1,
              "padding": 15,
              "paddingLeft": 15,
            }
          }
          testID="ItemComponent.ToucableOpacity"
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 15,
              }
            }
          >
            T15B.FRK
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
